Update Weave and Training reference generation tooling and content

mdlinville opened the pull request:
## Description
### Fix Weave reference documentation structure

- Removed extra 'weave' directory level from Python and TypeScript SDK paths
- Updated generation scripts to output directly to:
  - /weave/reference/python-sdk/ (not /weave/reference/python-sdk/weave/)
  - /weave/reference/typescript-sdk/ (not /weave/reference/typescript-sdk/weave/)
- Moved existing files into place
- Updated docs.json TOC entries to reflect correct paths
- Updated update_weave_toc.py script to use correct paths

### Fix landing page generation for Weave reference docs

- Updated Python SDK generation to output root 'weave' module to python-sdk.mdx
- Updated TypeScript SDK generation to create typescript-sdk.mdx instead of index.mdx
- Removed index.mdx files from SDK directories (using landing pages instead)
- Fixed TOC entries to reference landing pages (e.g., weave/reference/python-sdk)
  not index files (e.g., weave/reference/python-sdk/index)
- This follows Mintlify's convention where landing pages are at parent level
- Moved weave/reference/index.mdx to weave/reference.mdx (parent-level landing page)
- Updated TOC entry from 'weave/reference/index' to 'weave/reference'

### Fix broken links in Weave reference landing pages

- Updated typescript-sdk.mdx landing page links to use ./typescript-sdk/ prefix
- Fixed TypeScript SDK files linking to ../index to link to ../ (landing page)
- Updated generation scripts to generate correct paths for landing pages
- Landing pages now correctly reference their subdirectories with proper paths

### Fix Service API landing page links with fully qualified URLs

- Replaced relative links with https://docs.wandb.ai/ URLs
- This is necessary because Service API pages are dynamically generated
  from OpenAPI spec and don't exist as static files
- Fully qualified URLs will work correctly on the deployed site

### Fix more links manually

### Update Weave reference generation automation

- Add a script and action to sync the Service API spec to the repo
- Automate maintenance of Available Endpoints section of landing page
- Automate fully qualifying of deep Service API links from narrative pages
- Reorganize scripts

### Handle Training OpenAPI spec the same way
To prevent intermittent `502` errors processing the training spec remotely:
- Add scripts to download the OpenAPI spec and maintain the landing page
- Add Github Action to run the scripts on demand and cut a draft PR
- Update docs.json to point to the spec in the repo

## Testing
- [x] `mint dev` succeeds
- [x] `mint broken-links` is clean
- [x] Python scripts work locally and create expected output 
- [x] PR deploy and link-rot checks are clean

mdlinville committed:
Fix Weave reference documentation structure

- Removed extra 'weave' directory level from Python and TypeScript SDK paths
- Updated generation scripts to output directly to:
  - /weave/reference/python-sdk/ (not /weave/reference/python-sdk/weave/)
  - /weave/reference/typescript-sdk/ (not /weave/reference/typescript-sdk/weave/)
- Moved existing files into place
- Updated docs.json TOC entries to reflect correct paths
- Updated update_weave_toc.py script to use correct paths

Fix landing page generation for Weave reference docs

- Updated Python SDK generation to output root 'weave' module to python-sdk.mdx
- Updated TypeScript SDK generation to create typescript-sdk.mdx instead of index.mdx
- Removed index.mdx files from SDK directories (using landing pages instead)
- Fixed TOC entries to reference landing pages (e.g., weave/reference/python-sdk)
  not index files (e.g., weave/reference/python-sdk/index)
- This follows Mintlify's convention where landing pages are at parent level
- Moved weave/reference/index.mdx to weave/reference.mdx (parent-level landing page)
- Updated TOC entry from 'weave/reference/index' to 'weave/reference'

Fix broken links in Weave reference landing pages

- Updated typescript-sdk.mdx landing page links to use ./typescript-sdk/ prefix
- Fixed TypeScript SDK files linking to ../index to link to ../ (landing page)
- Updated generation scripts to generate correct paths for landing pages
- Landing pages now correctly reference their subdirectories with proper paths

Fix Service API landing page links with fully qualified URLs

- Replaced relative links with https://docs.wandb.ai/ URLs
- This is necessary because Service API pages are dynamically generated
  from OpenAPI spec and don't exist as static files
- Fully qualified URLs will work correctly on the deployed site

Fix more links manually

mdlinville committed:
Update Weave reference generation automation

- Add a script and action to sync the Service API spec to the repo
- Automate maintenance of Available Endpoints section of landing page
- Automate fully qualifying of deep Service API links from narrative pages

dbrian57 committed:
Fixes Weave links (#1759)

## Description
This PR fixes a Weave quickstart link and updates some copy on the Weave
index page.

dbrian57 committed:
fixed some broken links

mdlinville commented on weave/cookbooks/dspy_prompt_optimization.mdx:
```diff
@@ -39,15 +39,15 @@ os.environ["OPENAI_API_KEY"] = api_key
 ## Enable Tracking using Weave
 
-Weave is currently integrated with DSPy, and including [`weave.init`](https://docs.wandb.ai/guides/weave/tracking) at the start of our code lets us automatically trace our DSPy functions which can be explored in the Weave UI. Check out the [Weave integration docs for DSPy](https://docs.wandb.ai/guides/weave/integrations/dspy) to learn more.
+Weave is currently integrated with DSPy, and including [`weave.init`](/weave/reference/python-sdk/trace/weave_client#method-init) at the start of our code lets us automatically trace our DSPy functions which can be explored in the Weave UI. Check out the [Weave integration docs for DSPy](https://docs.wandb.ai/guides/weave/integrations/dspy) to learn more.
```
```suggestion
Weave is currently integrated with DSPy, and including [`weave.init`](/weave/reference/python-sdk/trace/weave_client#method-init) at the start of our code lets us automatically trace our DSPy functions which can be explored in the Weave UI. Check out the [Weave integration docs for DSPy](/weave/guides/integrations/dspy) to learn more.
```
I'll actually fix this elsewhere, just showing you what the link should be.

ngrayluna reviewed (dismissed):
Can you organize the weave specific generation scripts into a subdirctory? Brownie points if you add a README.md within said subdir that explains the entrypoint of these scripts, how they work.

mdlinville committed:
Reorganize reference generation scripts, rename workflow, update README

mdlinville commented:
> Can you organize the weave specific generation scripts into a subdirctory? Brownie points if you add a README.md within said subdir that explains the entrypoint of these scripts, how they work.

Done, and updated the existing README that is together with these scripts.

mdlinville committed:
Remove unused script

mdlinville committed:
Really fix dspy link

mdlinville committed:
Handle Training OpenAPI spec the same way as Service API

- scripts/reference-generation/training/: Scripts to sync the spec and maintain API landing page
- training/api-reference/openapi.json: Spec fetched using the relevant script
- training/api-reference.mdx: Added Available Endpoints section populated using the relevant script
- .github/workflows/update-training-api.yml: Github action that runs the scripts on demand and creates a draft PR
